Mathlib/Data/Finset/Image.lean

```diff
@@ -767,4 +767,24 @@ theorem finsetCongr_toEmbedding (e : α ≃ β) :
     e.finsetCongr.toEmbedding = (Finset.mapEmbedding e.toEmbedding).toEmbedding :=
   rfl
 
+/-- Given a predicate `p : α → Prop`, produces an equivalence between
+  `{s : Finset α // ∀ a ∈ s, p a}` and `Finset {a : α // p a}`. -/
+protected def finsetSubtypeCongr (p : α → Prop) :
+    {s : Finset α // ∀ a ∈ s, p a} ≃ (Finset {a : α // p a}) where
```

Comment:
I wonder if it's more helpful to give this equivalence in the opposite direction: I'd generally prefer to have a set together with the hypothesis that each of its elements satisfies a certain condition to a set of a subtype...